sig-architecture/api_changes.md

```diff
@@ -0,0 +1,71 @@
+
+### How to add a new version
+
+1. Pick a conversion strategy. Since custom resource objects need to be able to be served at both versions, that means they will sometimes be served at a different version than their storage version. In order for this to be possible, the custom resource objects must sometimes be converted between the version they are stored at and the version they are served at. If the conversion involves schema changes and requires custom logic, a conversion webhook should be used.
+2. If using conversion webhooks, create and deploy the conversion webhook. See the [Webhook conversion](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-conversion) for more details.
+3. Update the CustomResourceDefinition to include the new version in the `spec.versions` list with `served:true`. Also, set `spec.conversion` field to the selected conversion strategy. If using a conversion webhook, configure `spec.conversion.webhookClientConfig` field to call the webhook.
+
+Once the new version is added, clients may incrementally migrate to the new version. It is perfectly safe for some clients to use the old version while others use the new version.
+
+### Migrate stored objects to the new version
+
+Although it is safe for clients to use both the old and new version before, during and after upgrading the objects to a new stored version, but we need to deprecate and drop the old version support, for those API objects created in the old version before we introduce the new version, we need to migration the stored object in the APIServer yo new version.
+
+We have two options to acheived this:
+
+Option 1: Use the [Storage Version Migrator](https://github.com/kubernetes-sigs/kube-storage-version-migrator)
+Option 2: Manually upgrade the existing objects to a new stored version
+
+We prefer to option 1 because the migration process with option 1 is totally transparent to users, while option 2 need users' assistent and may introduce service interrupt.
+
+In addition, OCP 4.6+ will install the [Storage Version Migrator](https://github.com/openshift/kube-storage-version-migrator) be dafault. All we need to do is creating a migration request by the `StorageVersionMigration` resource similar to below:
```

Comment:
let's remove this sentence. I think we do not need to mention this

Reply:
done